### PR TITLE
#174 & #194 Support UTF-8 Encoding in String Method API

### PR DESCRIPTION
**Pull request by Maxwell-Alexius**
This PR aims to support the details stated in #174 and #194.

For future string API optimizations, it deserves another PR. So it is very welcome to help enhance or refactor String API methods!

For now, Goby lang support multilingual by using plain `rune` and `string` conversion.

### CODE REVIEW

File: vm/string.go [hunk 1]
```diff
@@ -288,6 +288,8 @@ func builtinStringInstanceMethods() []*BuiltInMethodObject {
 			// "Hello\nWorld"[5] # => "\n"
 			// "Hello"[-1]       # => "o"
 			// "Hello"[-6]       # => nil
+			// "Hello😊"[5]      # => "😊"
+			// "Hello😊"[-1]     # => "😊"
```
Comment (hachi8833) [inline vm/string.go]: Suggestion: checking '4-byte UTF-8 codepoints' such as '🍣 ' or '🍺' as well is better.
Ref: http://mysqlserverteam.com/sushi-beer-an-introduction-of-utf8-support-in-mysql-8-0/

Comment (Maxwell-Alexius) [inline vm/string.go]: Got it!

**Comment by st0012** [comment]
I'm not very familiar with encoding, so I just took a look at test cases and this LGTM. @hachi8833 you can merge it when you think it's ready.

**Comment by hachi8833** [comment]
As far as I see, `<=>` [line 225](https://github.com/Maxwell-Alexius/goby/blob/a9c82b93b8266c4c63d9fe3b1c262bb633012be9/vm/string.go#L225) should also be conscious of rune.

LGTM for the rest. 😄

**Comment by hachi8833** [comment]
I noticed that I can merge and then Maxwell can fix `<=>` after that.
Merging...

**Pull request merged**